src/Compilers/CSharp/Test/Symbol/Symbols/LocalFunctionTests.cs

```diff
@@ -102,5 +102,92 @@ void M()
             var staticLocal = semanticModel.GetDeclaredSymbol(localsSyntax[0]).GetSymbol<MethodSymbol>();
             Assert.False(staticLocal.RequiresInstanceReceiver);
         }
+
+        [Fact]
+        public void PartialStaticLocalFunction()
+        {
+            CreateCompilation("""
+                public class C
+                {
+                    public void M()
+                    {
+                        partial static void local() { }
+                    }
+                }
+                """).VerifyDiagnostics(
+                // (5,9): error CS0103: The name 'partial' does not exist in the current context
+                //         partial static void local() { }
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "partial").WithArguments("partial").WithLocation(5, 9),
+                // (5,17): error CS1002: ; expected
+                //         partial static void local() { }
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "static").WithLocation(5, 17),
+                // (5,29): warning CS8321: The local function 'local' is declared but never used
+                //         partial static void local() { }
+                Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "local").WithArguments("local").WithLocation(5, 29));
+        }
+
+        [Fact]
+        public void StaticPartialLocalFunction()
```

Comment:
This test serves the same purpose as the one I commented on for lambdas, but for this code path:

https://github.com/dotnet/roslyn/blob/7693a6fec1fb518981df205bf784aaf9d4b8995c/src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs#L47-L49